### ISSUE TITLE
Possible Optimization in Training FastSurferCNN/VINN

### ISSUE THREAD

**Issue by ndahiya3** [Deep-MI/FastSurfer#287]
## Question/Support Request

I am trying to use FastsurferCNN* architecture (my own implementation) to train with my own dataset. I noticed high GPU memory usage. Particularly, in the fastsurfer CNN code:
https://github.com/Deep-MI/FastSurfer/blob/stable/FastSurferCNN/models/sub_module.py

# First Maxout/Addition
```
x0_bn = torch.unsqueeze(x, 4) # Original input --> weighted with attention map 1
x1_bn = torch.unsqueeze(x1_bn, 4) # RF 3x3 --> weighted with attention map 2
x1 = torch.cat((x1_bn, x0_bn), dim=4)  # Concatenate along the 5th dimension NB x C x H x W x F
x1_max, _ = torch.max(x1, 4)
```
Instead of the above if I simply do:
`x1_max = torch.max(x0_bn, x1_bn) (without unsqueezing)`
much less GPU memory is used and I can increase my batch size.

Can someone confirm if there is anything wrong in directly using torch.max??

**Comment by LeHenschel** [comment]
Nice, thanks for reporting this. I think the operations should indeed be replaceable (quick test also gives me the same results for both ways). The version of max working on two tensors directly (torch.maximum) was not available before torch v 1.7 which is why we have the unsqueeze version in the code (original FastSurfer was using torch 1.2 I think).

**Comment by LeHenschel** [comment]
Added in Pull request #292. Closing this issue now.
